src/routes/version_creation.rs

```diff
@@ -0,0 +1,239 @@
+            let body_url = format!("data/{}/changelogs/{}/body.md", mod_id, version_id);
+
+            let uploaded_text = file_host
+                .upload_file(
+                    "text/plain",
+                    &body_url,
+                    version_create_data.version_body.clone().into_bytes(),
+                )
+                .await?;
+
+            uploaded_files.push(UploadedFile {
+                file_id: uploaded_text.file_id.clone(),
+                file_name: uploaded_text.file_name.clone(),
+            });
+
+            // TODO: do a real lookup for the channels
+            let release_channel = match version_create_data.release_channel {
+                VersionType::Release => models::ChannelId(0),
+                VersionType::Beta => models::ChannelId(2),
+                VersionType::Alpha => models::ChannelId(4),
+            };
+
+            version_builder = Some(VersionBuilder {
+                version_id: version_id.into(),
+                mod_id: models::ModId(mod_id.0 as i64),
+                name: version_create_data.version_title.clone(),
+                version_number: version_create_data.version_number.clone(),
+                changelog_url: Some(format!("{}/{}", cdn_url, body_url)),
+                files: Vec::with_capacity(1),
+                dependencies: version_create_data
+                    .dependencies
+                    .iter()
+                    .map(|x| (*x).into())
+                    .collect::<Vec<_>>(),
+                // TODO: add game_versions and loaders info
+                game_versions: vec![],
+                loaders: vec![],
+                release_channel,
+            });
+
+            continue;
+        }
+
+        let file_name = content_disposition.get_filename().ok_or_else(|| {
+            CreateError::MissingValueError("Missing content file name".to_string())
+        })?;
+        let file_extension = if let Some(last_period) = file_name.rfind('.') {
+            file_name.get((last_period + 1)..).unwrap_or("")
+        } else {
+            return Err(CreateError::MissingValueError(
+                "Missing content file extension".to_string(),
+            ));
+        };
+
+        if &*file_extension == "jar" {
+            let version = version_builder.as_mut().ok_or_else(|| {
+                CreateError::InvalidInput(String::from("`data` field must come before file fields"))
+            })?;
+
+            let mut data = Vec::new();
+            while let Some(chunk) = field.next().await {
+                data.extend_from_slice(&chunk.map_err(CreateError::MultipartError)?);
+            }
+
+            let upload_data = file_host
+                .upload_file(
+                    "application/java-archive",
+                    &format!(
+                        "{}/{}/{}",
+                        version.mod_id.0, version.version_number, file_name
```

Comment:
This should be just `version.mod_id` so that it is formatted as base62.  While using just the integer works, it isn't consistent with the rest of the ids.